framework/ldtest/filter_test.go

```diff
@@ -94,4 +94,6 @@ func TestRegexFilters(t *testing.T) {
 func TestAutoEscapeTestRegex(t *testing.T) {
 	assert.Equal(t, "hello \\(yes\\)", autoEscapeTestRegex("hello (yes)"))
 	assert.Equal(t, "hello \\(yes\\)", autoEscapeTestRegex("hello \\(yes\\)"))
+	assert.Equal(t, "hello \\[yes\\], a \\[test\\]", autoEscapeTestRegex("hello [yes], a [test]"))
+	assert.Equal(t, "hello \\[yes\\], a \\[test\\]", autoEscapeTestRegex("hello \\[yes\\], a \\[test\\]"))
```

Comment:
Wanted to ensure the escape would happen to all of the matched characters in a string, hence the repeat. Could add this to the `()` tests as well.